tests/test_cli.py

```diff
@@ -443,3 +456,46 @@ def test_fail_if_invalid_env_provided_in_run(
             str(result.exception)
             == f"Invalid env-var: {env_var}, expected format: KEY=VALUE"
         )
+
+
+@pytest.mark.parametrize(
+    "on_job_scheduled",
+    (
+        "bad_str_format",
+        "unexisting_module:func",
+        "tests.job_schedule_callback.existing_module:unexisting_attr",
+        "tests.job_schedule_callback.existing_module:attr_is_not_callable"
+        "tests.job_schedule_callback.existing_module:func_has_wrong_number_or_args",
+    ),
+    ids=(
+        "no_module",
+        "no_attr",
+        "not_func",
+        "wrong_nb_arg",
+    ),
+)
+def test_fail_if_invalid_on_job_scheduled_provided_in_run(
+    patched_kedro_package,
+    cli_context,
+    dummy_pipeline,
+    tmp_path: Path,
+    on_job_scheduled: str,
+):
+    create_kedro_conf_dirs(tmp_path)
+    with patch.dict(
+        "kedro.framework.project.pipelines", {"__default__": dummy_pipeline}
+    ), patch.object(Path, "cwd", return_value=tmp_path), patch(
+        "kedro_azureml.client.MLClient"
+    ) as ml_client_patched, patch(
+        "kedro_azureml.auth.utils.DefaultAzureCredential"
+    ), patch.dict(
+        os.environ, {"AZURE_STORAGE_ACCOUNT_KEY": "dummy_key"}
+    ):
+        ml_client = ml_client_patched.from_config()
+        ml_client.jobs.stream.side_effect = ValueError()
+
+        runner = CliRunner()
+        result = runner.invoke(
```

Comment:
Can you modify this test so that the callback is also mocked?
Pseudo-code:
```python
with #... other patches,
    patch("dummy.module:callback") as cb:
        # ...
```
and then assert that `cb.assert_called_once()`?